generators/app/index.js

```diff
@@ -0,0 +1,342 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+const Generator = require('yeoman-generator');
+const path = require('path');
+const Bundle = require("../../package.json");
+const Log4js = require('log4js');
+const logger = Log4js.getLogger('generator-core-golang-gin');
+const fs = require('fs');
+const helpers = require('../../lib/helpers.js');
+const swaggerize = require('ibm-openapi-support');
+const OPTION_BLUEMIX = 'bluemix';
+const OPTION_SPEC = 'spec';
+
+module.exports = class extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+
+    // Bluemix option for YaaS integration
+    this.argument(OPTION_BLUEMIX, {
+      desc: 'Option for deploying with Bluemix. Stringified JSON.',
+      required: false,
+      hide: true,
+      type: String
+    });
+
+    // Spec as json
+    this.argument(OPTION_SPEC, {
+      desc: 'The generator specification. Stringified JSON.',
+      required: false,
+      hide: true,
+      type: String
+    });
+  }
+
+  initializing() {
+    this.interactiveMode = false;
+    this.options.openApiFileType = 'yaml'; // Default
+    let bluemixOk = this._sanitizeOption(this.options, OPTION_BLUEMIX);
+    this._sanitizeOption(this.options, OPTION_SPEC);
+
+    if (!bluemixOk || typeof this.options.bluemix.quiet == "undefined" || !this.options.bluemix.quiet) {
+      logger.info("Package info ::", Bundle.name, Bundle.version);
+    }
+
+    // Set defaults
+    this.options.applicationType = this.options.applicationType || 'BLANK';
+    this.options.addServices = this.options.addServices || false;
+    this.options.bluemix = this.options.bluemix || {}
+    this.options.bluemix.backendPlatform = this.options.bluemix.backendPlatform || 'GO'
+
+    // Prompt the user if there is no bluemix object present
+    if (!bluemixOk) {
+      this.interactiveMode = true;
+    }
+
+    if (!this.interactiveMode) {
+      let appName = this.options.bluemix.name || this.options.spec.appname;
+      this.options.appName = helpers.sanitizeAppName(appName);
+
+      if (this.options.spec && this.options.spec.applicationType) {
+        this.options.applicationType = this.options.spec.applicationType;
+      }
+
+      this.options.parsedSwagger = undefined;
+
+      // If a swagger file is provided
+      if (
+        this.options.bluemix &&
+        this.options.bluemix.openApiServers &&
+        this.options.bluemix.openApiServers[0].spec
+      ) {
+        let openApiDocumentBytes =
+          typeof this.options.bluemix.openApiServers[0].spec === 'object'
+            ? JSON.stringify(this.options.bluemix.openApiServers[0].spec)
+            : this.options.bluemix.openApiServers[0].spec;
+
+        this._parseSwagger(openApiDocumentBytes);
+      }
+    }
+  }
+
+  prompting() {
```

Comment:
we're moving the prompting to the generator-goserver aren't we?

Reply:
All the prompts will be upfront on the go-server generator, but we still want to have prompts here in case a user wants to run this generator by itself ( it still needs to be able to work as a normal yeoman generator). However, when called from go-server or if given a bluemix json file, the prompts on the generator core-golang-gin are always skipped.